src/main/scala/org/broadinstitute/dsde/workbench/sam/api/UserRoutes.scala

```diff
@@ -20,61 +21,64 @@ trait UserRoutes extends UserInfoDirectives {
 
   def userRoutes: server.Route =
     pathPrefix("user") {
-      requireUserInfo { userInfo =>
-        (pathPrefix("v1") | pathEndOrSingleSlash){
+        (pathPrefix("v1") | pathEndOrSingleSlash) {
           pathEndOrSingleSlash {
             post {
-              complete {
-                userService.createUser(WorkbenchUser(userInfo.userId, userInfo.userEmail)).map(userStatus => StatusCodes.Created -> userStatus)
-              }
-            } ~
-            get {
-              parameter("userDetailsOnly".?) { userDetailsOnly =>
-                complete {
-                  userService.getUserStatus(userInfo.userId, userDetailsOnly.exists(_.equalsIgnoreCase("true"))).map { statusOption =>
-                    statusOption.map { status =>
-                      StatusCodes.OK -> Option(status)
-                    }.getOrElse(StatusCodes.NotFound -> None)
-                  }
-                }
-              }
-            }
-          }
-        } ~
-        pathPrefix("v2") {
-          pathPrefix("self") {
-            pathEndOrSingleSlash {
-              post {
+              requireCreateUser { createUser =>
```

Comment:
I don't think `newUserInfo` makes sense because it's not a new `UserInfo`...it's just an instance of `CreateWorkbenchUserAPI` case class...I'm up for other name suggestions tho

Reply:
I guess see my other comment on the naming of `CreateWorkbenchUserAPI` then.  I think if we can agree on what we want the final name for that class to be, it will inform a better variable name here.